✨(backend) create backend docker image

madmatah opened the pull request:
## Purpose

The official moodlnet backend docker image is based on the s6
process manager (via the s6-overlay project). It cannot be ran with a
non-privilegied user out of the box, due to the s6-overlay v1.22.1.0
that is used to build the image.

In addition to that, the image embeds a nginx server + the Erlang BEAM
application, supervised by s6. We want to have a container that run
_only_ the BEAM application.

And as a bonus, we prefer to use a debian based distribution instead
of alpine.

The goal of this docker image is to address these concerns.

## Proposal

- [x] Create a new moodlenet backend docker image
- [x] Update nginx configuration

madmatah committed:
✨(backend) create backend docker image

The official moodlnet backend docker image is based on the the s6
process manager (via the s6-overlay project). It cannot be ran with a
non-privilegied user out of the box, due to the s6-overlay v1.22.1.0
that is used to build the image.

In addition to that, the image embeds a nginx server + the Erlang BEAM
application, supervised by s6. We want to have a container that run
_only_ the BEAM application.

And as a bonus, we prefer to use a debian based distrubution instead
of alpine.

The goal of this docker image is to address these concerns.

jmaupetit commented on Makefile:
```diff
@@ -10,10 +13,20 @@ bootstrap: \
+.PHONY: build-back
+
+build-front: ## Build frontend image
+	@$(COMPOSE) build app
```
This suggests that the `app` service should be renamed `frontend` instead?

jmaupetit commented on docker/images/backend/Dockerfile:
```diff
@@ -0,0 +1,91 @@
+		rm -rf /var/lib/apt/lists/*
+
+RUN echo "$LANG UTF-8" >> /etc/locale.gen && \
+		locale-gen && \
```
weird indentation

madmatah replied:
Thanks, my editor indentation settings were... weird too :sweat_smile:

jmaupetit commented on docker/images/backend/Dockerfile:
```diff
@@ -0,0 +1,91 @@
+# dependencies for comeonin
+RUN apt-get update && \
+		apt-get install -y build-essential cmake curl git rustc cargo && \
+		rm -rf /var/lib/apt/lists/*
```
weird indentation

jmaupetit commented on docker/images/backend/Dockerfile:
```diff
@@ -0,0 +1,91 @@
+
+ENV APP_NAME=${APP_NAME} APP_VSN=${APP_VSN} APP_REVISION=${APP_VSN}-${APP_BUILD}
+
+WORKDIR /opt/app
```
So you are part of the `/opt` team... Why not choosing `/app` instead?

madmatah committed:
🔧(nginx) add missing route in nginx configuration

Some routes (starting with /.well-known/) served by moodlenet backend
were not proxied by nginx configuration.
